### DISCUSSION TITLE
[2024-05-13] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#9923] [Weekly Demo Day]
- James: Improve Table Viz handling of Numbers #9793
- James: Handle URL encoding automatically in query string. #9823
- James: Restructure `SQLite_Details`. #9832
- James: Small tweaks from Ned, Mark, Cass and Steve's feedback. #9859
- James: Undo change to autoscoping. #9881
- James: Fix the group colours for consistent colouring. #9893
- James: Error messages for `rename_columns` and `Vector.duplicates` #9917
- Radosław: Prepare for Cloud Environment variable name change #9821
- Radosław: Audit Logs for Postgres connections opened through a data link #9873
- Radosław: Followup improvements to the license review tool #9895
- Radosław: Update endpoint name after it was updated in the Cloud #9912
- Radosław: WIP: Creating datalinks in the Cloud programmatically #9673 
- Greg: Add `Vector.build_multiple` and handle dataflow error and warning propagation in `build` and `build_multiple` #9766
- Greg: Convert most remaining uses of `Vector.new_builder` to `Vector.build` #9891
- Greg: Combine builders for `Vector.build` and `Vector.new_builder` #9921
- Greg: Combine builders for `Vector.build` and `Vector.new_builder` #9922
- Ad: Add Ignored_Nothing_Values #9770
- Ad: Make Table.should_equal and Column.should_equal consider NaN equal #9799
- Ad: Refactor add row number #9822
- Ad: Fix Ordering Test #9836
- Ad: Make excel writer work for all types #9846
- Ad: Make table.Running return integer typed columns for min/max #9853
- Ad: Add table running variance skew sd and kurtosis #9854
- Ad: Add random icon #9877
- Ad: Add data cleanse component #9879
- Ad: Add missing tests #9897
- Jaroslav: Removing Truffle API dependency from `runtime-compiler` project #9785
- Jaroslav: Rewriting org.enso.runner.Main to Java #9810
- Jaroslav: Report non-zero frgaal exit code as an error #9834
- Jaroslav: Pretending WithWarnings isException #9840
- Jaroslav: Error on indents #9855
- Jaroslav: Syntax error shall contain location of the error #9864
- Jaroslav: Importing (some of) Standard.Base works from NI runner #9866
- Jaroslav: Reduce set of JDK modules required by Enso #9868
- Jaroslav: Language server needs java.desktop and java.se modules #9885
- Hubert: WIP: Extract mutable builder from `ModuleScope` #9914
- Hubert: Allow for profiling startup in dockerized setting #9865
- Hubert: Log timings of FS operations #9908
- Hubert: Compile-time `shapeless` in `runtime-compiler` #9805
- Hubert: Compile parser using musl #9806
- Hubert: Improve SuggestionsRepo initialization #9808
- Hubert: Minor follow up to #9528 #9841
- Hubert: Workaround slow responses with retries #9858
- Hubert: Don't recompute IR children #9915
- Hubert: Abort jobs as early as possible #9927 
- Hubert: WIP: Initial Bazel integration #9777 [on hold]
- Pavel: [WIP] Add logging for UnresolvedMethod bug #9763
- Pavel: Run engine benchmarks with increased stack size #9839
- Pavel: Conflicting extension methods result in compilation failure #9844
- Pavel: from conversion propagates dataflow error #9856
- Pavel: Shadowing identifiers produces warnings #9863
- Dmitry: Autoscoping should not escape True and False #9797
- Dmitry: Fix polyglot tree serialization #9852
- Dmitry: Start Ydoc with the language server #9862

**Comment by radeusgd** [comment]
Showing the audit logs from a short session of using an audited connection:

> ![image](https://private-user-images.githubusercontent.com/1436948/329192130-4719d5bb-22ec-47d5-b5eb-ee81dde10045.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2MDczNDgsIm5iZiI6MTcxNTYwNzA0OCwicGF0aCI6Ii8xNDM2OTQ4LzMyOTE5MjEzMC00NzE5ZDViYi0yMmVjLTQ3ZDUtYjVlYi1lZTgxZGRlMTAwNDUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MTNUMTMzMDQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDU2ZGRmMjJkYzdhN2U1MTFhYTRmYTdmMjY3ODhhODQ1ZmJkMmRhODQwYzk3NDkzMTAzNjY0MDllNGMwNTE5ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.h8kJ3bmQCkQoEsAKrjPyFAvxOBlPC4zML_y4nzeeWPs)
> 
> Report from OpenSearch logs, sorted by the sequence number. I highlighted a block corresponding to a table update rows transaction.
